linalg/operator.hpp

```diff
@@ -112,6 +113,10 @@ class Operator
    {
       return GetProlongation(); // Assume square unless specialized
    }
+   /** @brief Prolongation operator from linear algebra (linear system) vectors,
+       to output vectors for the operator, including only processor-local portion.
+       `NULL` means identity. */
+   virtual const Operator *GetLocalOutputProlongation() const { return NULL; }
```

Comment:
The way I think about this operator is that it assembles with `SetValues` instead of `AddValues`. You need to enforce that in two places, once at the element-assembly level (by using `SetValues`) and once at the parallel assembly level (by simply not writing on dofs you do not own, which means using a "local" matrix/operator so you only write on your own dofs). This is probably equivalent to your `R_test` left-inverse way of thinking, but I'm trying to explain where these names/documentation comes from. In particular, the `Operator` this returns is in fact processor-local.

I recall having this discussion with Veselin either on Slack or GitHub, and he suggested these "local" names, but I can't find it now and I could be remembering wrong.

Reply:
Thanks Andrew, I think we are on the same page.

Assembling is just the action of `P_test^T`, while "setting" is the action of `R_test`. This is easy to see in the Boolean conforming case, but also holds in more general cases, e.g. nonconforming AMR.

Are you saying that `LocalOutputProlongation` above is just the local portion of `R_test`? In that case I'm confused by the name "prolongation", this is really a restriction.

Reply:
`LocalOutputProlongation` is the local portion of `R_test^T`, it's shaped like a prolongation but I guess it's constructed more like `R_test`. In addition to clarifying the documentation (which I will do), do you have a suggestion for the name of the method itself? `GetOutputProlongation` already exists, and adding `Local` seemed like a reasonable way to distinguish this method from that one.

Reply:
Ok, I understand now.

I personally prefer a name like `GetOutputRestrictionTranspose`, but I'd like to understand something first -- why do we need another object, can't we just use `ParFiniteElementSpace::GetRestrictionMatrix`?

Reply:
Well, for one thing `ParFiniteElementSpace::GetRestrictionMatrix` returns a `SparseMatrix*`, and the whole point of this PR is to do things matrix-free. You could view this as another reason for the name, we build on `(Device)ConformingProlongationOperator` for the matrix-free action of these operators. I guess one option is to change the interface for `GetRestrictionMatrix()` to return an `Operator`, but that seems invasive.

It might be best to schedule a technical meeting about all this. I don't object to changing names etc., but there are reasons for what @dylan-copeland  and I did.